pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -162,23 +158,47 @@ func (c *QuorumGuardController) ensureEtcdGuard(ctx context.Context, recorder ev
 	return nil
 }
 
-// ensureEtcdGuardDeployment if etcd quorum guard deployment doesn't exist or was changed - apply it
-func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, replicaCount int32, recorder events.Recorder) error {
+// ensureEtcdGuardDeployment if etcd quorum guard deployment doesn't exist or was changed - apply it.
+func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, recorder events.Recorder) error {
+	replicaCount, err := c.getMastersReplicaCount(ctx)
+	if err != nil {
+		return err
+	}
 
 	if c.etcdQuorumGuard == nil {
 		c.etcdQuorumGuard = resourceread.ReadDeploymentV1OrDie(etcd_assets.MustAsset("etcd/quorumguard-deployment.yaml"))
 	}
 	c.etcdQuorumGuard.Spec.Replicas = &replicaCount
-	// use image from release payload
+	// Use image from release payload.
 	c.etcdQuorumGuard.Spec.Template.Spec.Containers[0].Image = c.cliImagePullSpec
 
-	// if restart occurred, we will apply etcd guard deployment but if it is the same, nothing will happen
+	// Set pod affinity only in HA topology mode.
+	affinity := c.etcdQuorumGuard.Spec.Template.Spec.Affinity
+	affinity.PodAffinity = &corev1.PodAffinity{
```

Comment:
Consider just appending to the `RequiredDuringSchedulingIgnoredDuringExecution` list rather than overriding the entirety of `PodAffinity` to allow other affinities to be specified in bindata. Or alternatively error/crash if other affinites are defined in bindata so they don't get silently wiped

Reply:
makes sense I added a note to the bindata spec and an error if podAffinity exists.